packages/kubernetes-common/src/types.ts

```diff
@@ -0,0 +1,208 @@
+ */
+
+import {
+  ExtensionsV1beta1Ingress,
+  V1ConfigMap,
+  V1Deployment,
+  V1HorizontalPodAutoscaler,
+  V1Pod,
+  V1ReplicaSet,
+  V1Service,
+} from '@kubernetes/client-node';
+import { Entity } from '@backstage/catalog-model';
+
+export interface ClusterDetails {
+  name: string;
+  url: string;
+  authProvider: string;
+  serviceAccountToken?: string | undefined;
+}
+
+export interface KubernetesRequestBody {
+  auth?: {
+    google?: string;
+  };
+  entity: Entity;
+}
+
+export interface ClusterObjects {
+  cluster: { name: string };
+  resources: FetchResponse[];
+  errors: KubernetesFetchError[];
+}
+
+export interface ObjectsByEntityResponse {
+  items: ClusterObjects[];
+}
+
+export interface FetchResponseWrapper {
+  errors: KubernetesFetchError[];
+  responses: FetchResponse[];
+}
+
+export type FetchResponse =
+  | PodFetchResponse
+  | ServiceFetchResponse
+  | ConfigMapFetchResponse
+  | DeploymentFetchResponse
+  | ReplicaSetsFetchResponse
+  | HorizontalPodAutoscalersFetchResponse
+  | IngressesFetchResponse
+  | CustomResourceFetchResponse;
+
+// TODO fairly sure there's a easier way to do this
+
+export type KubernetesObjectTypes =
+  | 'pods'
+  | 'services'
+  | 'configmaps'
+  | 'deployments'
+  | 'replicasets'
+  | 'horizontalpodautoscalers'
+  | 'ingresses'
+  | 'customresources';
+
+export interface PodFetchResponse {
+  type: 'pods';
+  resources: Array<V1Pod>;
+}
+
+export interface ServiceFetchResponse {
+  type: 'services';
+  resources: Array<V1Service>;
+}
+
+export interface ConfigMapFetchResponse {
+  type: 'configmaps';
+  resources: Array<V1ConfigMap>;
+}
+
+export interface DeploymentFetchResponse {
+  type: 'deployments';
+  resources: Array<V1Deployment>;
+}
+
+export interface ReplicaSetsFetchResponse {
+  type: 'replicasets';
+  resources: Array<V1ReplicaSet>;
+}
+
+export interface HorizontalPodAutoscalersFetchResponse {
+  type: 'horizontalpodautoscalers';
+  resources: Array<V1HorizontalPodAutoscaler>;
+}
+
+export interface IngressesFetchResponse {
+  type: 'ingresses';
+  resources: Array<ExtensionsV1beta1Ingress>;
+}
+
+export interface CustomResourceFetchResponse {
+  type: 'customresources';
+  resources: Array<any>;
+}
+
+export interface ObjectFetchParams {
+  serviceId: string;
+  clusterDetails: ClusterDetails;
+  objectTypesToFetch: Set<KubernetesObjectTypes>;
+  labelSelector: string;
+  customResources: CustomResource[];
+}
+
+// Fetches information from a kubernetes cluster using the cluster details object
+// to target a specific cluster
+export interface KubernetesFetcher {
+  fetchObjectsForService(
+    params: ObjectFetchParams,
+  ): Promise<FetchResponseWrapper>;
+}
+
+// Used to locate which cluster(s) a service is running on
+export interface KubernetesServiceLocator {
+  getClustersByServiceId(serviceId: string): Promise<ClusterDetails[]>;
+}
+
+// Used to load cluster details from different sources
+export interface KubernetesClustersSupplier {
```

Comment:
Would say it's best to keep backend-only types like this one in the backend package